From a82d529db5c995f550fecc9ce0fa7394b6067b8b Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alex=20R=C3=B8nne=20Petersen?= <alex@alexrp.com>
Date: Wed, 3 Sep 2025 11:23:44 +0200
Subject: [PATCH 071/133] std.debug: disable stack traces on loongarch
X-Developer-Signature: v=1; a=openpgp-sha256; l=703; i=xtex@astrafall.org;
 h=from:subject; bh=a9xil4Cj9XvvIRkw8ASPBqiIt/1nq9MVJUNBLYHQKA0=;
 b=owGbwMvMwCW2U4Ij7wZL9ETG02pJDJn78lhaY+Kif6c9UPORO5gTquf7bm14cfOp8k0+xvpNl
 +U+p/N2TGRhEONisBRTZCkybPBm1UnnF11WLgszh5UJZIi0SAMDAwMDCwNfbmJeqZGOkZ6ptqGe
 oaGOkY4RAxenAEz1t25GhuePzfqqzFc0vBH5vuQoM9d+4Q1rHI/fuzwrQOrG/GkvWqwZGQ7Pf1o
 W93Vud8iHSdWcnkX9wctXcHIKTDz5xMQ98/86WXYA
X-Developer-Key: i=xtex@astrafall.org; a=openpgp;
 fpr=7231804B052C670F15A6771DB918086ED8045B91

Observed to ~randomly crash during FP-based unwinding.

The path forward here will be DWARF-based unwinding.

(cherry picked from commit d8d0a3e5c77ccce33699d6c11a850d383a83c40d)
---
 lib/std/debug.zig | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/lib/std/debug.zig b/lib/std/debug.zig
index 26ea76a9d5eb..797735d4b9a4 100644
--- a/lib/std/debug.zig
+++ b/lib/std/debug.zig
@@ -171,6 +171,8 @@ pub const runtime_safety = switch (builtin.mode) {
 pub const sys_can_stack_trace = switch (builtin.cpu.arch) {
     // Observed to go into an infinite loop.
     // TODO: Make this work.
+    .loongarch32,
+    .loongarch64,
     .mips,
     .mipsel,
     .mips64,
-- 
2.52.0

